botocore/waiter.py

```diff
@@ -287,9 +287,11 @@ def __init__(self, name, config, operation_method):
     def wait(self, **kwargs):
         acceptors = list(self.config.acceptors)
         current_state = 'waiting'
-        sleep_amount = self.config.delay
+        # pop the invocation specific config
+        config = kwargs.pop('WaiterConfig', {})
+        sleep_amount = config.get('delay', self.config.delay)
+        max_attempts = config.get('maxAttempts', self.config.max_attempts)
```

Comment:
I did this to be consistent with the `PaginateConfig` formatting. The keys are capitalized words there. The waiter config uses mixed case keys. No matter what the key is called it will break consistency in some way.